### ISSUE TITLE
Cross compiling eventmachine for Linux platforms.

### ISSUE THREAD

**Issue by karthikirco** [eventmachine/eventmachine#851]
I am trying to cross compile a ruby gem "Eventmachine". My target is an ARM Linux machine and my host is a x86_64 Linux machine. I found out his link this and it talks about cross compiling for Windows platform and not much about Linux platforms. Is cross compiling ruby gems supported on Linux platforms ?

I already have ruby and openssl cross compiled in Yocto build which are the dependencies of the ruby gem Eventmachine. It will be also helpful if anyone can point out how to point the cross compiler to use the cross compiled Ruby and Openssl in instead of host machine's.

Thank you for your time.

**Comment by sodabrew** [comment]
Yes, cross compiling is possible, but not something I've done except for the Windows builds. See https://github.com/rake-compiler/rake-compiler that the gem this gem uses to cross-compile to Windows. It should be possible to set up a cross-compile environment to another platform as well, via https://github.com/rake-compiler/rake-compiler-dock (using a Docker image that provides ARM cross-compilers).

**Comment by sodabrew** [comment]
Fedora 7 was released 11 years ago, is that really your distro version? https://fedoraproject.org/wiki/Releases/7

Unfortunately I'm not deeply familiar with the cross-compile process; I've relied on the tools that wrap this!

**Comment by karthikirco** [comment]
Let me double check the Fedora version. Anyway i don't think my fedora falls in Docker's range. What kind of tools are you talking about ?
